extensions/Worker.Extensions.Http.AspNetCore.Analyzers/RegistrationExpectedInASPNetIntegration.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore
+{
+    /// <summary>
+    /// Analyzer to verify whether expected registration is present for ASP.NET Core Integration.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class RegistrationExpectedInASPNetIntegration : DiagnosticAnalyzer
+    {
+        /// Diagnostics supported by the analyzer
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(DiagnosticDescriptors.CorrectRegistrationExpectedInAspNetIntegration);
+
+        private const string ExpectedRegistrationMethod = "ConfigureFunctionsWebApplication";
+        private const string IncorrectRegistrationMethod = "ConfigureFunctionsWorkerDefaults";
+
+        /// Initialization method
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
+            context.RegisterSymbolAction(AnalyzeMethod, SymbolKind.Method);
+        }
+
+        private static void AnalyzeMethod(SymbolAnalysisContext context)
+        {
+            var symbol = (IMethodSymbol)context.Symbol;
+
+            if (!IsMainMethod(symbol))
+            {
+                return;
+            }
+
+            var syntaxReference = symbol.DeclaringSyntaxReferences.FirstOrDefault();
+            var root = syntaxReference?.SyntaxTree.GetRoot();
+            var methodCallExpressions = root?.DescendantNodes().OfType<InvocationExpressionSyntax>();
+
+            var expectedMethodInvocationPresent = methodCallExpressions?.Any(invocation => (invocation.Expression as MemberAccessExpressionSyntax)?.Name.Identifier.ValueText == ExpectedRegistrationMethod);
+
+            var incorrectMethodCallExpressions = methodCallExpressions?.Where(invocation => (invocation.Expression as MemberAccessExpressionSyntax)?.Name.Identifier.ValueText == IncorrectRegistrationMethod);
+            var incorrectMethodInvocationPresent = incorrectMethodCallExpressions?.Any();
+
+            if (!(bool)incorrectMethodInvocationPresent)
```

Comment:
This casting expression will throw if `incorrectMethodInvocationPresent` has `null`. Consider making `incorrectMethodInvocationPresent` bool type instead of `bool?`.

Reply:
Because we have `incorrectMethodCallExpressions?.Any()`, it will have to be `bool?`. What I do in this case is null-coalesce for the null case:

Not sure what the behavior should be when `null`, but something like this (or `false` instead of `true`)
``` CSharp
if (!(incorrectMethodInvocationPresent ?? true))
```